src/setup_payload/ManualSetupPayloadGenerator.cpp

```diff
@@ -87,56 +87,78 @@ static uint32_t chunk3PayloadRepresentation(const SetupPayload & payload)
     return result;
 }
 
-// TODO: issue #3663 - Unbounded stack in src/setup_payload
-#pragma GCC diagnostic push
-#if !defined(__clang__)
-#pragma GCC diagnostic ignored "-Wstack-usage="
-#endif
-#pragma GCC diagnostic ignored "-Wvla"
-
-static std::string decimalStringWithPadding(uint32_t number, int minLength)
+static void decimalStringWithPadding(char * buf, size_t len, uint32_t number)
 {
-    char buf[minLength + 1];
-    snprintf(buf, sizeof(buf), "%0*" PRIu32, minLength, number);
-    return std::string(buf);
+    snprintf(buf, len + 1, "%0*" PRIu32, static_cast<int>(len), number);
```

Comment:
why is this len + 1 ? "natural" would seem that we should pass in a buffer + sizeofbuffer not buffer + size of buffer - 1.

Could we use a MutableByteSpan instead?

Reply:
Ok yeah that would make more sense. Also changed to use spans.